cygnus-ngsi/src/main/java/com/telefonica/iot/cygnus/sinks/NGSIPostgreSQLSink.java

```diff
@@ -507,18 +507,19 @@ public void aggregate(NGSIEvent cygnusEvent) {
                 LOGGER.debug("[" + getName() + "] Processing context attribute (name=" + attrName + ", type="
                         + attrType + ")");
 
-                // create part of the column with the current attribute (a.k.a. a column)
                 if (attrNativeTypes) {
-                    if (attrType.equals("Number")) {
-                        column += "," + attrValue + ",'"  + attrMetadata + "'";
-                    } else {
-                        if (attrValue == null || attrValue.equals("")) {
-                            attrValue = "NULL";
+                    if (contextAttribute.getValue().isJsonNull()) {
+                        column += ",NULL,'" + attrMetadata + "'";
```

Comment:
One aditional remark. It should be nice to have an aditional flag in config like "NullValuesForMissingAttributes". This flag is needed to enable backward compatibility for those clients that used to insert "" for missing attributes in batches.